rdt-open-reader/src/testCovidIndonesia/java/io/ona/rdt/robolectric/interactor/CovidPatientVisitFragmentInteractorTest.java

```diff
@@ -0,0 +1,57 @@
+package io.ona.rdt.robolectric.interactor;
+
+import org.joda.time.DateTime;
+import org.joda.time.format.DateTimeFormat;
+import org.junit.Assert;
+import org.junit.Test;
+import org.mockito.ArgumentMatchers;
+import org.mockito.Mockito;
+import org.powermock.reflect.Whitebox;
+import org.robolectric.util.ReflectionHelpers;
+import org.smartregister.domain.Event;
+import org.smartregister.domain.db.EventClient;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import io.ona.rdt.domain.Visit;
+import io.ona.rdt.interactor.CovidPatientVisitFragmentInteractor;
+import io.ona.rdt.repository.PatientHistoryRepository;
+import io.ona.rdt.robolectric.RobolectricTest;
+
+public class CovidPatientVisitFragmentInteractorTest extends RobolectricTest {
+
+    private CovidPatientVisitFragmentInteractor interactor;
+
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+        interactor = new CovidPatientVisitFragmentInteractor();
+    }
+
+    @Test
+    public void testGetPatientVisits() {
+        PatientHistoryRepository patientHistoryRepository = Mockito.mock(PatientHistoryRepository.class);
+        ReflectionHelpers.setField(interactor, "patientHistoryRepository", patientHistoryRepository);
+
+        int listSize = 1;
+        List<EventClient> eventClientList = new ArrayList<>(listSize);
+        Event event = new Event();
+        event.setEventDate(DateTime.now());
+        eventClientList.add(new EventClient(event));
+
+        Mockito.when(patientHistoryRepository.getEventsByUniqueDate(ArgumentMatchers.anyString())).thenReturn(eventClientList);
+
+        List<Visit> visitList = interactor.getPatientVisits("");
+        Assert.assertEquals(listSize, visitList.size());
+        Assert.assertEquals("Visit 1", visitList.get(listSize - 1).getVisitName());
+        Assert.assertNull(visitList.get(listSize - 1).getDateOfVisit());
+    }
+
+    @Test
+    public void testFormatDate() throws Exception {
+        DateTime dateTime = DateTimeFormat.forPattern("yyyy-MM-dd HH:mm:ss").parseDateTime("2020-09-03 02:00:00");
+        String formattedDate = Whitebox.invokeMethod(interactor, "formatDate", dateTime);
+        Assert.assertNull(formattedDate);
```

Comment:
Any reason this should return null?

Reply:
because it was not working even i use the correct formatting. checkout this https://github.com/robolectric/robolectric/issues/1030

Reply:
Null means that there's an error somewhere. It should return a non-null date otherwise.

Reply:
yes, it should return a non-null date but the problem was this method https://github.com/onaio/rdt-standard/blob/5f386b8520888432389947125cf3f4eeb224f8cc/rdt-open-reader/src/main/java/io/ona/rdt/util/Utils.java#L101
it works fine with the real environment and throws an exception in the test environment

Reply:
could you look-out this link https://github.com/robolectric/robolectric/issues/1030

Reply:
what is the real and test environment?

Reply:
What exactly is the error when it fails?

Reply:
> what is the real and test environment?

I mean when it works fine on the device but failed in the test

Reply:
> What exactly is the error when it fails?

java.text.ParseException: Unparseable date: "2020-07-20T09:00:00.000+05:00"java.text.ParseException: Unparseable date: "2020-07-20T09:00:00.000+05:00"

Reply:
Which device? You mean locally? The unit tests don't run on a device.

Reply:
i mean this method works fine in application but throw an exception when a call from the test